pandas/core/indexing.py

```diff
@@ -2023,6 +2023,16 @@ def ravel(i):
                         return ser._values.copy()
                     return ser.reindex(ax)._values
 
+        elif is_integer(indexer) and self.ndim == 1:
```

Comment:
can any of this be shared with the is_scalar branch on L2036?  The `_get_axis(1)` there looks wrong, and it looks sketchy that `indexer` doesnt actually get used there

Reply:
We could generalize the code, but this makes it quite hard to read. Have already tried this.

The second case is when setting a series for a row in a DataFrame while the first case is setting a series (with length one obviously) as a row of a series

Reply:
makes sense, thanks.

can at least the is_scalar check below be tightened to is_integer?

Reply:
Yeah, this is certainly possible